pkgs/development/libraries/bash/bash-preexec/default.nix

```diff
@@ -0,0 +1,40 @@
+{ stdenvNoCC, lib, fetchFromGitHub, bats }:
+
+let version = "0.4.1";
+in stdenvNoCC.mkDerivation {
+  pname = "bash-preexec";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "rcaloras";
+    repo = "bash-preexec";
+    rev = version;
+    sha256 = "062iigps285628p710i7vh7kmgra5gahq9qiwj7rxir167lg0ggw";
+  };
+
+  checkInputs = [ bats ];
+
+  dontConfigure = true;
+  doCheck = true;
+  dontBuild = true;
+
+  patchPhase = ''
+    # Needed since the tests expect that HISTCONTROL is set.
+    sed -i '/setup()/a HISTCONTROL=""' test/bash-preexec.bats
+  '';
+
+  checkPhase = ''
+    bats test
+  '';
+
+  installPhase = ''
+    install -Dm755 $src/bash-preexec.sh $out/share/bash/bash-preexec.sh
+  '';
+
+  meta = with lib; {
+    description = "preexec and precmd functions for Bash just like Zsh";
+    license = licenses.mit;
+    homepage = "https://github.com/rcaloras/bash-preexec";
+    maintainers = [ maintainers.rycee ];
```

Comment:
would you mind adding the platforms meta attribute ?
reading the github repo, it seems that `platforms.unix` would be good here.
I wonder why darwin is not being marked for building this package.

Reply:
Good point, I missed adding the platform attribute (it must have been missing from the package I copied the code from 😉). Added now.